### ISSUE TITLE
A beta dataset doesn't show correct number of fastq data files in Browser

### ISSUE THREAD

**Issue by malloryfreeberg** [DataBiosphere/data-browser#470]
For the [1.3 Million Brain Cells from E18 Mice](https://prod.data.humancellatlas.org/explore/projects?filter=%5B%7B%22facetName%22%3A%22project%22%2C%22terms%22%3A%5B%221M+Neurons%22%5D%7D%5D) beta dataset, there are listed "0" fastq data files in the Browser (see screenshot). This project actually has 16,377 fastq files. 

![screen shot 2019-02-13 at 11 43 04](https://user-images.githubusercontent.com/10993501/52709549-f854b400-2f84-11e9-9716-7fe08163bdc0.png)

┆Issue Number: HCAB-385

**Comment by theathorn** [comment]
@hannes-ucsc Is this an Azul issue?

**Comment by hannes-ucsc** [comment]
Took a cursory look and it appears as though the data files in that project are described by metadata as having `"file_format": "fastq"` even though their file extension is `fastq.gz`.  The Data Browser seems to look for `fastq.gz` in the `fileTypeSummary` of the Azul response but display that in a column labelled `fastq`. That's why the correct number is displayed, but in the `Other` column. The Data Browser should combine the `fastq` and `fastq.gz` entries of `fileTypeSummary` since both formats conceptually represent a "fastq" file. The latter is just a compressed form of the former.

However, this is also metadata inaccuracy. If the file extension is `fastq.gz`, the file_format field should reflect that accurately since `.fastq` and `.fastq.gz` are different formats. Inaccuracies like that are should be avoided with a validation rule, and the project needs to be reingested with accurate metadata. If we let this slide, everyone downstream (external and internal users alike) will have to incorporate business logic into their programs to work around this inaccuracy. 

We can also eliminate the `file_format` property and have Azul inspect the actual file extension directly. With that logic in place, the higher-level `content_description` field that the metadata team is planning on implementing could replace `file_format`, not augment it.

Assigning back to @malloryfreeberg and @NoopDog for feedback.

**Comment by malloryfreeberg** [comment]
Nice sleuthing! 

As I'm thinking through the `fastq` vs. `fastq.gz` issue, I think what we really want in this field is `fastq`, but the format of the file is indeed fastq, it just also happens to be compressed. I think in general we want to ignore indicating compression in the `file_format` field, as that doesn't explain how the file itself is formatted. 

I think you're right that after we merge the `content_description` [PR](https://github.com/HumanCellAtlas/metadata-schema/pull/752), that field could replace the `file_format` field. 

I will make a note to wranglers that we should update all current spreadsheets with fastq for `file_format`, although I don't know of any specific plans to re-ingest any datasets at the moment (I can haz AUDR?)

**Comment by hannes-ucsc** [comment]
@malloryfreeberg, if we officially switch from `fastq.gz` to `fastq` now, we'd just be increasing the confusion. We'd have 14 projects using `fastq.gz` to describe `fastq.gz` files, and the rest would use `fastq`. Can't we just stick with `fastq.gz` and fix this one outlier project by reingesting it? 

An ontology term would the best solution as it could express that `fastq.gz` is an enconding of the `fastq` format. Then again, there already are standards for this: HTTP and MIME.

```
Content-Type: application/x-fastq
Content-Encoding: gzip
```

To summarize, I would strongly urge to not increase the entropy by switching to `fastq` now. I would also recommend adopting existing standards like MIME. The DSS already has support for content type.

PS: Asking wranglers to do something is good, implementing a validation rule is better.

**Comment by malloryfreeberg** [comment]
@hannes-ucsc one step ahead of you on using ontologies (EDAM) for the `file_format` field ;) (https://github.com/HumanCellAtlas/metadata-schema/issues/812).

The EDAM ontology _does_ include a term for "fastq", but not for "fastq.gz". While it's not impossible to get terms added to the ontology, I want to be sure that we do it for the right reasons. I still think the correct way to refer to the format of a sequencing data file is "fastq". The fact that it was compressed using gzip prior to file transfer is not _really_ relevant to how the file is formatted.

Would like to hear other's opinions, maybe on the next metadata call.

**Comment by hannes-ucsc** [comment]
I think it *is* relevant, @malloryfreeberg. Anyone processing the (meta)data programmatically would not be helped by `file_format` being `fastq` if the file is actually a `fastq.gz`. They would still have to look at the file extension to decide how to actually open and load the file. If they have to look at the file extension anyway, they can get all of the necessary information out of it (the fact that it's a fastq and that it's gzipped) and would not even need to consider `file_format` at all. 

What is the point of this field if it doesn't accurately describe how a file needs to be handled?

Two alternative suggestions: Either add `fastq.gz` to EDAM or add a `file_encoding` field to metadata schema. For the encoding I would draw upon MIME, not some ontology specific to bioinformatics (EDAM doesn't even have zip, gzip or bzip2). I would have said the same for the ontology on `file_format` but MIME doesn't have `fastq` so I guess EDAM is the only option for that.

**Comment by theathorn** [comment]
For now, Data Browser will count fastq and fastq.gz files as "fastq" for data Count column and exclude fastq.gz from "Other" count. File facets and download flow to remain unchanged.

**Comment by diekhans** [comment]
I think we really need to rigorously define what is meant by file_format.   Is it the bytes on disk or the data the program reads.  I  don't think of fastq.gz as a different format than fastq.  Many programs transparently handle compressed or uncompressed data, others need special handling for compressed data

I think it would be clearer to have the concepts of data format and compression format separate.  Maybe it is data_format and file_format, or the suggest file_encoding. 

This is what @hannes-ucsc MIME example does; separates the concepts/

Until we have the ability to update metadata, MHO is we should just hack around it until we can update metadata.

**Comment by theathorn** [comment]
In Prod select "1.3 Million Brain Cells from E18 Mice" and it shows 16,377 Fastqs but the "File" drop-down shows 1 fastq.

**Comment by theathorn** [comment]
@hannes-ucsc wrote: In a response to /repository/projects, the counts in termFacets represent number of projects, not number of files. In this case, for example, the termFacets counts means that there are 16 projects with fastq.gz files and 1 project with fastq files.

**Comment by NoopDog** [comment]
ug thx

**Comment by NoopDog** [comment]
Created https://app.zenhub.com/workspaces/dcp-backlogs-5ac7bcf9465cb172b77760d9/issues/humancellatlas/data-browser/489 to make the units in the facets more visible.
